.github/workflows/release.yml

```diff
@@ -0,0 +1,116 @@
+    steps:
+      - uses: actions/checkout@v5
+
+      - name: Install pinned Rust toolchain
+        run: |
+          rustup toolchain install ${RUST_TOOLCHAIN} --profile minimal --target wasm32v1-none
+          rustup default ${RUST_TOOLCHAIN}
+        env:
+          RUSTUP_TOOLCHAIN: ${{ env.RUST_TOOLCHAIN }}
+
+      - name: Cache Rust dependencies
+        uses: stellar/actions/rust-cache@main
+
+      - run: rustup target add wasm32v1-none
+        env:
+          RUSTUP_TOOLCHAIN: ${{ env.RUST_TOOLCHAIN }}
+
+      - run: cargo +${RUST_TOOLCHAIN} version
+
+      - name: Set up env vars
+        run: |
+          echo "WASM_FILE=contracts/target/wasm32v1-none/release/web_auth.wasm" >> $GITHUB_ENV
+
+          if [ -n "${{ github.event.inputs.release_name }}" ]; then
+            echo "TAG_NAME=${{ github.event.inputs.release_name }}" >> $GITHUB_ENV
+          else
+            echo "TAG_NAME=${{ github.ref_name }}" >> $GITHUB_ENV
+          fi
+
+      - uses: stellar/stellar-cli@v23.3.0
+        with:
+          version: 23.3.0
+
+      - name: Build contract
+        working-directory: contracts
+        run: |
+          stellar contract build \
+            --meta source_repo=github:${{ github.repository }}
+
+          stellar contract optimize --wasm target/wasm32v1-none/release/web_auth.wasm
+          cp target/wasm32v1-none/release/web_auth_verify.optimized.wasm target/wasm32v1-none/release/web_auth.wasm
+
+      - name: Upload to Artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: web_auth.wasm
+          path: ${{ env.WASM_FILE }}
+
+      - name: Build Attestation for Release
+        uses: actions/attest-build-provenance@v2.0.1
+        with:
+          subject-name: web_auth.wasm
+          subject-path: ${{ env.WASM_FILE }}
+
+      - name: Make a new Release
+        id: release
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const response = await github.rest.repos.createRelease({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              tag_name: '${{ env.TAG_NAME }}',
+              target_commitish: '${{ github.sha }}',
+              make_latest: 'true'
+            });
+
+            const { data } = response;
+            core.setOutput('release_id', data.id);
+
+      - name: Upload to Release
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const fs = require('fs');
+            const path = require('path');
+            await github.rest.repos.uploadReleaseAsset({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              release_id: '${{ steps.release.outputs.release_id }}',
+              name: path.basename('${{ env.WASM_FILE }}'),
+              data: fs.readFileSync('${{ env.WASM_FILE }}'),
+            });
```

Comment:
Something that you may want to consider doing is creating the release as a draft rather than marking it as latest, uploading the files, and then updating the release after that to be the latest. That way, before the files are connected to the release, there aren't notifications going out to anyone watching the repository.

Doing it that way would also allow Ops to configure this repository so that its releases are immutable once they are published. It's a GitHub feature we noticed recently and we haven't started using yet, but you can probably start using it on this repository and be a good test case for the security team.